crates/subspace-node/src/commands/run/domain.rs

```diff
@@ -28,25 +28,29 @@ use sc_service::Configuration;
 use sc_service::config::{ExecutorConfiguration, KeystoreConfig};
 use sc_transaction_pool_api::OffchainTransactionPoolFactory;
 use sc_utils::mpsc::{TracingUnboundedReceiver, TracingUnboundedSender};
-use sp_api::ProvideRuntimeApi;
+use sp_api::{ApiExt, ProvideRuntimeApi};
 use sp_blockchain::HeaderBackend;
 use sp_consensus_subspace::SubspaceApi;
 use sp_core::crypto::{AccountId32, SecretString};
-use sp_domains::{DomainId, DomainInstanceData, OperatorId, RuntimeType};
+use sp_domains::{DomainId, DomainInstanceData, DomainsApi, OperatorId, RuntimeType};
+use sp_runtime::traits::Block as BlockT;
 use std::collections::HashMap;
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 use std::sync::Arc;
 use subspace_runtime::RuntimeApi as CRuntimeApi;
 use subspace_runtime_primitives::opaque::Block as CBlock;
-use subspace_runtime_primitives::{
-    DOMAINS_BLOCK_PRUNING_DEPTH, DOMAINS_PRUNING_DEPTH_MULTIPLIER, HeaderFor,
-};
+use subspace_runtime_primitives::{DOMAINS_PRUNING_DEPTH_MULTIPLIER, HeaderFor};
 use subspace_service::FullClient as CFullClient;
 use tokio::sync::broadcast::Receiver;
 use tokio_stream::wrappers::errors::BroadcastStreamRecvError;
 use tracing::log::info;
 use tracing::warn;
 
+/// Domains Block pruning depth.
+/// This value is here because we this to set default state pruning and block pruning for clap arguments.
+/// Unfortunately clap does not support dynamic defaults.
+const DOMAINS_BLOCK_PRUNING_DEPTH: u32 = 14_400;
```

Comment:
One way to make defaults dynamic in clap is optional arguments. If it is `None`, set the dynamic default, otherwise, use the supplied value.

Reply:
The goal is show this number when then someone uses `--help` which is what I'm aiming for here